Marlin/Marlin_main.cpp

```diff
@@ -4087,14 +4087,21 @@ inline void gcode_G28(const bool always_home_all) {
 
       if (home_all || homeX || homeY) {
         // Raise Z before homing any other axes and z is not already high enough (never lower z)
-        destination[Z_AXIS] = Z_HOMING_HEIGHT;
-        if (destination[Z_AXIS] > current_position[Z_AXIS]) {
+        // We should never move more than Z_HOMING_HEIGHT distance.
+        #if DISABLED(MIN_SOFTWARE_ENDSTOP_Z)
+          if (current_position[Z_AXIS] < 0.0f)
```

Comment:
What is the relevance of SOFTWARE_ENDSTOP? Before homing they can't work and should not be active for that. And after once homed the problem is gone.

Reply:
Software endstops work before homing.  You can test this by turning on your printer and trying a G1 X-10.  Nothing will happen.

Reply:
And just to make this a little bit more complex.   I think this issue only shows up when the printer homes to Z-Max, right?    We can probably make it so what ever 'fix' we put in place doesn't disrupt anybody else.

Reply:
No, I believe that this is only an issue when the printer homes to Z-min.